constructor_assignment_lab.js

```diff
@@ -0,0 +1,200 @@
+// # Constructor Exercises
+
+// ## Question 1
+
+// a. Write a constructor function called `Person` that has 3 properties: a first name, a last name and a middle name. Create 2 instances of a `Person`. Print one of their first names.
+function Person(firstName, lastName, middleName) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.middleName = middleName;
+}
+Person.prototype.fullName = function() {console.log(this.firstName + " " + this.middleName + " " + this.lastName)}
+let brittany = new Person("Brittany", "Gaston", "Bernice")
+let jovanni = new Person("Jovanni", "Luna", "David")
+console.log(brittany.firstName)
+brittany.fullName()
+jovanni.fullName()
+
+// b. Write a prototype function in `Person` called `fullName` that will return a formatted string of an instance's full name. Call this method on both the instances you created in part a.
+
+
+// ## Question 2
+
+// a. Create a constructor function called `Book` that has properties `title`, `author` and `rating`. Create some instances of `Book`.
+function Book(title, author, rating) {
+    this.title = title;
+    this.author = author;
+    this.rating = rating;
+}
+Book.prototype.isGood = function() {return this.rating >= 7 ? true : false}
+let theGreatGatsby = new Book("The Great Gatsby", "F Scott Fitzgerald", 4)
+let ofMice = new Book("Of Mice and Men", "John Steinbeck", 7)
+let midsummer = new Book("A Midsummer Night's Dream", "William Shakespeare", 9)
+console.log(theGreatGatsby.isGood())
+// b. Add a prototype function to `Book` called `isGood` that returns `true` if its rating is greater than or equal to 7
+
+// ## Question 3
+function Dog(name, breed, mood, hungry) {
+    this.name = name;
+    this.breed = breed;
+    this.mood = mood;
+    this.hungry = hungry;
+}
+Dog.prototype.playfetch = function() {
+    this.hungry = true;
+    this.mood = "playful";
+    return "Ruff!"}
+Dog.prototype.feed = function() {
+    if(this.hungry === true){
```

Comment:
Because this.hungry evaluates to a boolean there is no need to do the === true. It's not wrong, just not needed.